homeassistant/components/shelly/bluetooth/__init__.py

```diff
@@ -0,0 +1,93 @@
+"""Bluetooth support for shelly."""
+from __future__ import annotations
+
+from typing import TypedDict
+
+from aioshelly.rpc_device import RpcDevice
+
+from homeassistant.components.bluetooth import (
+    HaBluetoothConnector,
+    async_get_advertisement_callback,
+    async_register_scanner,
+)
+from homeassistant.const import __version__ as HA_VERSION
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback as hass_callback
+from homeassistant.helpers.device_registry import format_mac
+
+from ..coordinator import ShellyRpcCoordinator
+from .const import (
+    BLE_CODE,
+    BLE_SCAN_RESULT_EVENT,
+    BLE_SCAN_RESULT_VERSION,
+    BLE_SCRIPT_NAME,
+    VAR_ACTIVE,
+    VAR_EVENT_TYPE,
+    VAR_HA_VERSION,
+    VAR_VERSION,
+)
+from .scanner import ShellyBLEScanner
+
+
+class ShellyScript(TypedDict, total=False):
+    """Shelly Script."""
+
+    id: int
+    name: str
+    enable: bool
+    running: bool
+
+
+async def _async_get_scripts_by_name(device: RpcDevice) -> dict[str, int]:
+    """Get scripts by name."""
+    data = await device.call_rpc("Script.List")
+    scripts: list[ShellyScript] = data["scripts"]
+    return {script["name"]: script["id"] for script in scripts}
+
```

Comment:
Is this something we should move into `aioshelly`?

Reply:
Yes